### PR TITLE
feat: [`Pay`] components

### PR DESCRIPTION
**Pull request by 0xAlec**
**What changed? Why?**
- add `PayButton`, `Pay`, and `PayStatus`
- add corresponding tests
- add `useGetPayStatus`
- modify `useCommerceContracts`callback to take an `address` as an argument

**Notes to reviewers**

**How has it been tested?**
playground PR next

### CODE REVIEW

File: src/pay/components/PayProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+            ) || coinbaseWallet({ preference: 'smartWalletOnly' }),
+        });
+        connectedAddress = accounts[0];
+        connectedChainId = _connectedChainId;
+      }
+
+      // This shouldn't ever happen, but added this to make Typescript happy
+      /* v8 ignore next 3 */
+      if (!connectedAddress) {
+        return;
```
Comment (0xAlec) [inline src/pay/components/PayProvider.tsx:180]: I think maybe we can add an `updateLifecycleStatus` error message here, but almost always there will be `connectedAddress` returned from `connectAsync`

File: src/pay/components/Pay.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    return null;
+  }
+
+  return (
+    <PayProvider
+      chargeHandler={chargeHandler}
+      onStatus={onStatus}
+      productId={productId}
+    >
+      <div className={cn(className, 'flex w-full flex-col gap-2')}>
```
Comment (alessey) [inline src/pay/components/Pay.tsx:25]: className last

File: src/pay/components/PayProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+    try {
+      // Open Coinbase Commerce receipt
+      if (lifecycleStatus.statusName === PAY_LIFECYCLESTATUS.SUCCESS) {
+        window.open(
+          `https://commerce.coinbase.com/pay/${chargeId}/receipt`,
+          '_blank',
+          'noopener,noreferrer',
+        );
+        return;
+      }
```
Comment (alessey) [inline src/pay/components/PayProvider.tsx:147]: may be more straightforward to have separate callbacks for handleFundingFlow and handleViewReceipt and then call different callbacks in the button based on the status

File: src/pay/components/PayProvider.tsx [hunk 4]
```diff
@@ -0,0 +1,281 @@
...
+
+      // Open keys.coinbase.com for payment
+      await writeContractsAsync({
+        contracts,
+      });
+    } catch (error) {
+      if (
+        (error as Error).message?.includes('User denied connection request')
+      ) {
+        setErrorMessage(USER_REJECTED_ERROR);
```
Comment (alessey) [inline src/pay/components/PayProvider.tsx:246]: shouldn't this also updateLifecycleStatus?

File: src/pay/components/PayProvider.tsx (hunk 3 above)
Comment (0xAlec) [reply src/pay/components/PayProvider.tsx:152]: I'll refactor this as a fast follow once we have USDC magic spend

Comment (0xAlec) [reply src/pay/components/PayProvider.tsx:152]: I don't think it makes sense to add a `handleFundingFlow.ts` just to delete it in a week or so

**Review by alessey — APPROVED** [review]

**Review by abcrane123 — APPROVED** [review]

**Pull request merged**